public/app/themes/clarity/inc/admin/prior-party/prior-party-banner-email.php

```diff
@@ -0,0 +1,455 @@
+        }
+
+        $args = [['dst' => false]];
+        if (!wp_next_scheduled('prior_party_banner_email_cron_hook', $args)) {
+            wp_schedule_event(strtotime('09:02:00'), 'daily', 'prior_party_banner_email_cron_hook', $args);
+        }
+
+        add_action('prior_party_banner_email_cron_hook', [$this, 'maybeSendEmails']);
+
+        // Load the data.
+        add_action('init', [$this, 'loadData'], 20);
+    }
+
+    /**
+     * Filter the ACF field to display the email digest content.
+     * 
+     * Make use of a message field to display the content of the email digest.
+     * 
+     * @param array $field The field array.
+     * 
+     * @return array The filtered field array.
+     */
+
+    public function placeholderField($field)
+    {
+        // Are we on the options page? 
+        // It's important not to filter on the ACF > Fields Groups > Setting page.
+        $screen = get_current_screen();
+        if (is_admin() && $screen->base === 'tools_page_prior-party-settings') {
+            $field['label'] = '';
+            ob_start();
+            $this->emailPage();
+            $field['message'] = ob_get_clean();
+            return $field;
+        }
+
+        return $field;
+    }
+
+    /**
+     * A function to convert a timestamp to a local date object.
+     * 
+     * @param int $timestamp The timestamp.
+     * 
+     * @return \DateTime
+     */
+
+    public function timestampToLocalDateObject(int $timestamp): \DateTime
+    {
+        $date_object = new \DateTime();
+        $date_object->setTimezone(new \DateTimeZone('Europe/London'));
+        $date_object->setTimestamp($timestamp);
+        return $date_object;
+    }
+
+    /**
+     * A timezone aware function that will send digest emails to those set in the Options page.
+     * 
+     * @param ?array $props An optional array containing the DST status.
+     * 
+     * @return void
+     */
+
+    public function maybeSendEmails(?array $props = []): void
+    {
+        // Is London currently observing DST?
+        $in_dst = strtotime("today 9:00 am Europe/London") !== strtotime("today 9:00 am UTC");
+
+        // If the current DST status doesn't match the props, return early.
+        if (isset($props['dst']) && $props['dst'] !== $in_dst) {
+            // It's 10am in the summer or 8am in the winter - don't send emails.
+            return;
+        }
+
+        $timestamp_from = strtotime('yesterday 9:00 Europe/London');
+        $timestamp_to = strtotime('today 9:00 Europe/London');
+        $email_content = $this->getEmailDigestByTimes($timestamp_from, $timestamp_to);
+
+        if (empty($email_content)) {
+            return;
+        }
+
+        // Was an email passed in manually - for testing purposes?
+        if (isset($props['recipient'])) {
+            wp_mail($props['recipient'], $email_content['subject'], $email_content['body']);
+            echo 'A test email was sent to email sent to ' . $props['recipient'];
```

Comment:
Hey @EarthlingDavey, I'm getting an error - possibly false-positive - from testing code that has made its way to production - I took these files from the main branch on Intranet.  Just a minor, but can we try to ensure that the code we use for testing doesn't go to prod? This will apply to debug stuff too. Let's chat if there are edge cases.

Can I remove this?